tf/caendr/modules/site/cloud-sql.tf

```diff
@@ -0,0 +1,72 @@
+resource "google_sql_database" "postgres_db_main" {
+  charset   = "UTF8"
+  collation = "en_US.UTF8"
+  instance  = google_sql_database_instance.postgres_instance.name
+  name      = var.module_site_vars.postgres_db_name
+  project   = var.google_cloud_vars.project_id
+}
+
+resource "google_sql_database" "postgres_db_stage" {
+  charset   = "UTF8"
+  collation = "en_US.UTF8"
+  instance  = google_sql_database_instance.postgres_instance.name
+  name      = var.module_site_vars.postgres_db_stage_name
```

Comment:
help me understand the naming here -- assuming the `cloud-sql` module is environment agnostic, are we defining two instances (postgres_dn_name and postgres_db_stage_name). Does `_stage` have a special meaning within a given environment?

Reply:
Updating the DB with update-or-replace queries can have issues when the external data sources being pulled have changes that break relationships in the data. Dan was just dropping the tables and then rebuilding them from scratch, but that has significant downtime, temporarily incomplete data, and if the task fails it needs to be rolled back. The stage DB is used for all the ETL operations, then dumped as a .sql backup in cloud storage and dropped. That .sql can be loaded much faster to the active db or the stage db can be used while the master db is being rebuilt. Since the only billing is for the instance and not the db, I figured this approach would reduce downtime without adding cost.